### PR TITLE
Cleaning up

### PR DESCRIPTION
**Pull request by jjship**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,57 +1,81 @@
...
-  const input = document.querySelector('input');
-  const controls = document.getElementsByClassName('controls');
-  const operators = document.getElementsByClassName('operators');
-  const nums = document.getElementsByClassName('nums');
-  let calculation = '';
-  const nested = document.getElementById('nested')
-  function isPoint(ch) { return (ch === ".");}
-  function isDigit(ch) { return /\d/.test(ch);}
-  function isOperator(ch) { return /\+|-|\*|\/|\^/.test(ch);}
+const buttonValues = [
```
[the remark this answers is not in the document]
Comment (MaciejReimann) [reply app.js]: also, for the fun of it, try to implement `createNaturalNumbersArray` function and use it here

File: app.js [hunk 2]
```diff
@@ -1,57 +1,81 @@
...
-    return result;
-  };
-       
-  
-  // digit => push ch to NB  
-  // decimal point => push ch to NB  
-  //operator => join NB contents as one Literal and push to result 
+})
 
+//defining Token class
```
Comment (MaciejReimann) [inline app.js]: this is not a class yet, this is a factory function

File: app.js [hunk 3]
```diff
@@ -1,57 +1,81 @@
...
+function Token(type, value) {   
+  this.type = type;   
+  this.value = value;
+}
 
+function tokenize(str) {  
+  var result=[]; //array of tokens
```
Comment (MaciejReimann) [inline app.js]: this is not an "array of tokens", this is the result

File: app.js [hunk 4]
```diff
@@ -1,57 +1,81 @@
...
-  };
-       
-  
-  // digit => push ch to NB  
-  // decimal point => push ch to NB  
-  //operator => join NB contents as one Literal and push to result 
+})
 
+//defining Token class
+function Token(type, value) {
```
Comment (MaciejReimann) [inline app.js]: unless you're gonna use this somewhere, this declaration should be within the scope of `tokenize`

Comment (MaciejReimann) [reply app.js]: (mainly for aesthetic reasons)

File: styles.css [hunk 5]
```diff
@@ -27,8 +27,57 @@
   background: #093DF6;
   padding:1em;
 }
-#zero {
-  grid-column: 1/3;;
+[id = '7'] {
```
Comment (MaciejReimann) [inline styles.css]: there must be a simpler way to make a grid of buttons

File: styles.css [hunk 6]
```diff
@@ -8,7 +8,7 @@
   display:grid;
   grid-gap:.5em;
   grid-template-columns: repeat(4,1fr);
-  grid-template-rows: repeat(5,minmax(40px, auto));
+  grid-template-rows: repeat(4,minmax(40px, auto));
 }
 .nested > div{
```
Comment (MaciejReimann) [inline styles.css]: Prettier!

File: app.js [hunk 7]
```diff
@@ -1,57 +1,81 @@
...
+      buffer.push(char);   
+    } else if (isOperator(char)) {   
+       //operator => join buffer contents as one Literal and push to result  
+      result.push(new Token("Literal", buffer.join('')));  
+      buffer =[];  
+      result.push(new Token("Operator", char));     
+    }   
+  });
+  
+  if (buffer) {result.push(new Token("Literal", buffer.join('')))};
```
Comment (MaciejReimann) [inline app.js]: redundant curly braces if inline

File: app.js [hunk 8]
```diff
@@ -1,57 +1,81 @@
...
+}
 
+function tokenize(str) {  
+  var result=[]; //array of tokens  
+  var buffer=[];
+  str=str.split("");  
+  str.forEach(function (char, idx) {    
+    if(isDigit(char)||isPoint(char)) {  
+      buffer.push(char);   
+    } else if (isOperator(char)) {
```
Comment (MaciejReimann) [inline app.js]: no need for `else if`

File: app.js [hunk 9]
```diff
@@ -1,57 +1,81 @@
...
 
+function isPoint(ch) { return (ch === ".");}
+function isDigit(ch) { return /\d/.test(ch);}
+function isOperator(ch) { return /\+|-|\*|\/|\^/.test(ch);}
 
-  nested.addEventListener('click', (e) => {
-    if (e.target.className === 'nums') {
-      let num = e.target.innerText;
-      calculation += num;
+function addButtons(array) {
```
Comment (MaciejReimann) [inline app.js]: it is `appendButtons` what it really is

Comment (MaciejReimann) [reply app.js]:
```
function appendButtonsToWrapper(buttonsArray, buttonsWrapper) { 
   // you'll know what to do 
}
```

File: app.js [hunk 10]
```diff
@@ -1,57 +1,81 @@
...
+  array.forEach((item) => {
+    let newButton = document.createElement("button");
+    newButton.className = item[0];
+    newButton.id = item[1];
+    newButton.innerHTML = item[1];
+    nested.appendChild(newButton);
+  });
+}
+addButtons(buttonValues);
```
[the remark this answers is not in the document]
Comment (jjship) [reply app.js]: but I need the buttons to be in place before I set the eventListeners, don't I?

File: styles.css (hunk 5 above)
Comment (jjship) [reply styles.css]: any hints? grid-area?

**Pull request merged**